app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java

```diff
@@ -199,22 +200,33 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     @MainThread
     private void handleSignInResponse(int resultCode, Intent data) {
-        if (resultCode == RESULT_OK) {
-            startActivity(SignedInActivity.createIntent(this, IdpResponse.fromResultIntent(data)));
-            finish();
-            return;
+        if (data != null) {
```

Comment:
There are a few things here:

- I'd like checking for RESULT_OK to be a top-level check.  That's our contract, not the nullity of `data`.
- When the resultCode is CANCELED we will have two quick snackbars right?  One because the IdpResponse contains UNKNOWN_ERROR and one because it was canceled.